lib/task.js

```diff
@@ -273,9 +275,43 @@ export class Task {
         .concat(oldEntrypoint);
     }
 
+    if(this.task.payload.features && this.task.payload.features.allowPtrace) {
+      // generate and load a new AppArmor profile specialized to this taskId
+      // allowing ptrace within the container
+      let aa_profile = await this.makePtraceAppArmorProfile(this.status.taskId);
+
+      // add the equivalent of --security-opt apparmor:docker-ptrace
+      let hc = procConfig.create.HostConfig
+      hc.SecurityOpt = hc.SecurityOpt || [];
+      hc.SecurityOpt.push("apparmor:" + aa_profile);
+    }
+
     return procConfig;
   }
 
+  async makePtraceAppArmorProfile(taskId) {
+    // copy and modify the docker-default profile
+    let profile = await fs.readFile("/etc/apparmor.d/docker");
+    let name = "worker-" + taskId;
+
+    // update the name of the profile
+    profile = profile.replace("profile docker-default", "profile " + name);
+
+    // add a rule allowing ptrace to other processes with this policy, at
+    // the very end
+    profile = profile.replace("\n}", "\n  ptrace (trace,read) peer=" + name + "\n}");
+
+    // write out the new profile
+    await fs.writeFile("/etc/apparmor.d/" + name, profile)
```

Comment:
nit: we probably don't need to worry about this since instances are not long lived, the profile is small, and the instances do not claim a lot of tasks, but these old apparmor profiles are not cleaned up after the tasks completes. 

Not something that I see as an immediate thing to clean up though.